### DISCUSSION TITLE
How can I determine if a Message Editor is readonly?

### DISCUSSION THREAD

**Discussion by cyal1** [PortSwigger/burp-extensions-montoya-api#71] [Q&A]
```java
public List<Component> provideMenuItems(ContextMenuEvent event) {
if (event.messageEditorRequestResponse().isPresent()) {
            MessageEditorHttpRequestResponse messageEditor = event.messageEditorRequestResponse().get();
//  Message Editor is readonly?
}
```
How can I determine if a Message Editor is readonly? Thanks.

**Comment by Hannah-PortSwigger** [comment]
You can find an example of an editor's read-only state being passed through [here](https://github.com/PortSwigger/burp-extensions-montoya-api-examples/blob/cfbef6db3761d32fbe6551349695c8ab673e794e/customrequesteditortab/src/main/java/example/customrequesteditortab/MyExtensionProvidedHttpRequestEditor.java#L48).

**Reply by cyal1** [reply]
I perform certain tasks using the context menu item, like some Content-Type conversion, need setRequest in editor. The determination is made within the context menu to check whether the current message editor is read-only, rather than implementing my own message editor.

**Reply by Hannah-PortSwigger** [reply]
If you're able to retrieve the context that the context menu has been called in, you could use that to determine whether a message editor is read-only.

Please find some ToolType examples below:
LOGGER: Read-only
PROXY: Likely to be read-only
REPEATER: Editable
SCANNER: Read-only
SUITE: Read-only
TARGET: Read-only
